maven/maven.bzl

```diff
@@ -174,35 +185,36 @@ def _generate_maven_repository_impl(ctx):
             artifact = artifacts.annotate(artifacts.parse_spec(spec))
             coordinates = "%s:%s" % (artifact.group_id, artifact.artifact_id)
             sets.add(processed_artifacts, coordinates)
-            maven_deps = _get_dependencies_from_pom_files(ctx, artifact)
-            maven_deps = [x for x in maven_deps if _should_include_dependency(x)]
-            found_artifacts = {}
-            bazel_deps = []
-            for dep in maven_deps:
-                found_artifacts[dep.coordinate] = dep
-                bazel_deps += [_convert_maven_dep(ctx.attr.name, dep)]
-            normalized_deps = [_normalize_target(x, group_path, package_target_substitutes) for x in bazel_deps]
-            unregistered = sets.difference(processed_artifacts, sets.copy_of(found_artifacts))
-            if bool(unregistered) and not bool(build_substitutes.get(coordinates)):
-                unregistered_deps = [
-                    poms.format_dependency(x)
-                    for x in maven_deps
-                    if sets.contains(unregistered, x.coordinate)
-                ]
-                fail("Some dependencies of %s were not pinned in the artifacts list:\n%s" % (
-                    spec,
-                    list(unregistered_deps),
-                ))
-            target_definitions += [
-                build_substitutes.get(
-                    coordinates,
+            snippet = build_snippets.get(coordinates, None)
+            if snippet:
```

Comment:
Note: There is a bit of a behavior change here, which is htat if you override the build snippet, it doesn't bother to go fetch the pom file.  This can actually fix a possible problem where you want to use the build snippet to fix a case of badly formed xml or something else that would crash.  Now, there won't even be a pom fetch if you've supplied a replacement snippet.

Reply:
nvm, I noted that in the PR description.